### PR TITLE
MCP-2247: fix NPE in BipClaimService

### PR DESCRIPTION
**Pull request by dimitri-amida**
## What was the problem?
BipClaimService fails with NPE when pdfResponse.reason is null.

Associated tickets or Slack threads:

- [MCP-2247](https://amida.atlassian.net/browse/MCP-0002247)

## How does this fix it?[^1]
Handle the null condition gracefully.

## How to test this PR
- Run the normal automatedClaimRoute in test or local. 
- Look in the database for the audit trail. There should be no exceptions.

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipClaimService.java [hunk 1]
```diff
@@ -169,6 +170,9 @@ public MasProcessingObject completeProcessing(MasProcessingObject payload) {
    */
   public FetchPdfResponse uploadPdf(FetchPdfResponse pdfResponse) {
     log.info("Uploading pdf for claim {}...", pdfResponse.getClaimSubmissionId());
+    if (pdfResponse.getPdfData() == null) {
+      throw new MasException("PDF Response does not contain any data");
+    }
     String filename = String.format("temp_evidence-%s.pdf", pdfResponse.getClaimSubmissionId());
```
Comment (wlin2019) [inline service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipClaimService.java:176]: Should MasException be specified with throws? Similarly, should the BipException thrown by bipApiService.uploadEvidence(...) be caught or thrown?

Comment (dimitri-amida) [reply service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipClaimService.java:174]: @wlin2019 good points. I changed to BipException and declared it with "throws"

File: app/src/main/java/gov/va/vro/config/SecurityConfig.java [hunk 2]
```diff
@@ -16,7 +15,6 @@
 import org.springframework.security.web.SecurityFilterChain;
 import org.springframework.security.web.authentication.HttpStatusEntryPoint;
 
-@Slf4j
```
Comment (dimitri-amida) [inline app/src/main/java/gov/va/vro/config/SecurityConfig.java:19]: not used

File: service/provider/src/main/java/gov/va/vro/service/provider/bip/BipException.java [hunk 3]
```diff
@@ -6,11 +6,6 @@
  * @author warren date 10/31/22
  */
 public class BipException extends RuntimeException {
-  private static final String BIP_ERROR = "VA BIP API access error.";
-
-  public BipException() {
```
Comment (dimitri-amida) [inline service/provider/src/main/java/gov/va/vro/service/provider/bip/BipException.java:11]: not used

**Review by rajesh-Krishnasamy — APPROVED** [review]

**Pull request merged**